core-bundle/src/Filesystem/FilesystemItemIterator.php

```diff
@@ -153,4 +187,32 @@ public function toArray(): array
     {
         return iterator_to_array($this, false);
     }
+
+    private static function sortByMediaTypePriority(FilesystemItem $a, FilesystemItem $b): int
+    {
+        if (!$a->isFile() && !$b->isFile()) {
+            return 0;
+        }
+
+        if (!$a->isFile() || !$b->isFile()) {
+            if ($b->isFile()) {
+                return 1;
+            }
+
+            return -1;
+        }
+
+        $sortOrderA = array_search($a->getMimeType(), self::MEDIA_TYPE_SORT_ORDER, true);
+        $sortOrderB = array_search($b->getMimeType(), self::MEDIA_TYPE_SORT_ORDER, true);
+
+        if (false === $sortOrderA) {
+            return 1;
+        }
+
+        if (false === $sortOrderB) {
+            return -1;
+        }
+
+        return $sortOrderB < $sortOrderA ? 1 : -1;
```

Comment:
Could condense this down a bit with spaceships. But probably very opinionated. Feeld free to ignore :slightly_smiling_face: 

```suggestion
            $aIsFile = $a->isFile();
        
            if(0 !== ($sort = ($b->isFile() <=> $aIsFile))) {
                return $sort;
            } else if(!$aIsFile) {
                return 0;
            }

            $sortOrderA = array_search($a->getMimeType(), MEDIA_TYPE_SORT_ORDER, true);
            $sortOrderB = array_search($b->getMimeType(), MEDIA_TYPE_SORT_ORDER, true);

            return ($sortOrderA === false ? PHP_INT_MAX : $sortOrderA) <=> ($sortOrderB === false ? PHP_INT_MAX : $sortOrderB);
```